core/src/main/scala/org/broadinstitute/dsde/rawls/model/DataReferenceModel.scala

```diff
@@ -13,7 +17,16 @@ object DataRepoSnapshotReference {
   }
 }
 
+object DataRepoSnapshotListFromDataReferenceList {
+  def apply(ref: DataReferenceList): DataRepoSnapshotList = {
+    val buffer = for (r <- ref.getResources.asScala) yield DataRepoSnapshotReference(r)
+    DataRepoSnapshotList(buffer.toList)
+  }
+}
+
```

Comment:
stylistically, I would not make a new object class for this. 

I would make this a companion object to `DataRepoSnapshotList`. I realize that in doing so, you introduce ambiguity on the `apply` methods ... so I'd rename your custom method  to `from`:
```
object DataRepoSnapshotList {
  def from(ref: DataReferenceList): DataRepoSnapshotList = {
    ...
```

Then in SnapshotService, change:
```
Future.successful(DataRepoSnapshotListFromDataReferenceList(ref))
```
to
```
Future.successful(DataRepoSnapshotList.from(ref))
```

Reply:
Ah! `companion object` was the term I was missing to google what was going on here, thanks!
My only motivation for making this a separate object class was avoiding the ambiguous `apply` issue - I wasn't sure if renaming the class or the `apply` method was a worse style violation, I'll switch this.